File: pr073/pr073.c
#ifdef _MSC_VER
    #define DLL_EXPORT __declspec(dllexport)
#else
    #define DLL_EXPORT
#endif

#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

DLL_EXPORT long run(void);
int main(void);
long gcd(long a, long b)
{
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

DLL_EXPORT long run(void)
{
    long ret;
    long x, y;
    ret = 0;
    for (x = 4; x <= 12000; x++)
        for (y = (long)(x/3)+1; y < (x >> 1) + (x & 1); y++)
            if (gcd(x, y) == 1)
                ret ++;
    return ret; 
}

int main(void)
{
    printf("%d\n", run());
    return 0;
}

File: pr073/pr073.py
#coding:utf8
def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)

def pr073():
    ret = 0
    for x in range(4, 12001):
        for y in range(x//3 + 1, x//2 + (x & 1)):
            if gcd(x, y) == 1:
                ret += 1
    return ret

def run():
    return pr073()

if __name__ == "__main__":
    print(run())

